src-electron/generator/helper-endpointconfig.js

```diff
@@ -112,6 +113,19 @@ function endpoint_fixed_parent_id_array(options) {
   return '{ ' + parentIds.join(', ') + ' }'
 }
 
+async function endpoint_composition() {
+  let dt
+  let db = this.global.db
+  for (const et of this.endpointTypes) {
+    dt = await queryDeviceType.selectDeviceTypeById(db, et.deviceTypeRef)
+  }
+  if (dt.composition != null) {
+    return '{' + dt.composition + '}'
+  } else {
+    return '0'
+  }
```

Comment:
This does not make sense to me.
- you are going through all endpoint types in the zap configuration
- getting device type from each endpoint(not sure what happens when there are multiple device types on an endpoint type)
- Returning the last endpoint's device type composition from the list of endpoints types you traversed.
- Where will this helper be used?

Reply:
Oh true I'll fix that thanks